libs/core/src/lib/injectors/mapped-injector.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  Inject,
+  Injectable,
+  InjectFlags,
+  InjectionToken,
+  Injector,
+  Provider,
+  Type,
+} from '@angular/core';
+
+import { InjectorMap } from '../types';
+
+export interface InjectorMapToken extends Array<InjectorMap> {}
+
+/**
+ * Multi-provider of {@link InjectorMap}
+ */
+export const INJECTOR_MAP_TOKEN = new InjectionToken<InjectorMapToken>(
+  'INJECTOR_MAP',
+);
+
+/**
+ * Helper to provide {@link INJECTOR_MAP_TOKEN}
+ */
+export function provideInjectorMap(map: InjectorMap): Provider {
+  return { provide: INJECTOR_MAP_TOKEN, useValue: map, multi: true };
+}
+
+/**
+ * Maps tokens to other tokens and then executes parent injector.
+ *
+ * NOT a Service!
+ * Use via {@link MappedInjectorFactory}
+ */
+export class MappedInjector implements Injector {
+  private injectorMaps: InjectorMap[];
```

Comment:
May be we can use struct like that `injectorMaps: {[token: string]: InjectorMap[];}`

Reply:
Will add caching for this

Reply:
Updated